src/components/AlertLine.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <li 
+    class="alert" 
+    :style="{ 'background-color': alert.backgroundColor }">
+    <div class="icon-container">
+      <img class="icon" :src="`img/${topic}.svg`" />
+    </div>
+    <div class="alert-message">{{ alert.message }}</div>
+    <div class="alert-date">{{ alert.time }}</div>
+    <a @click="toggleCollapse" href="#" class="icon-container">
+      <img v-if="collapsed" class="icon" src="img/down-arrow.svg" />
+      <img v-else class="icon" src="img/up-arrow.svg" />
+    </a>
+    <div v-if="!collapsed" class="alert-details">
+      <h3>Server Information</h3>
+      <ul class="server-info">
+        <li>FQDN: {{alert.fqdn}}</li>
+        <li>Hostname: {{alert.serverInfo.Hostname}}</li>
+        <li>ClusterID: {{alert.serverInfo.ClusterID}}</li>
+        <li>ServerID: {{alert.serverInfo.ClusterID}}</li>
+      </ul>
+    </div>
+    <div v-if="!collapsed" class="alert-buttons">
+      <button @click="snoozeAlert">Snooze Alert</button>
+      <button @click="openGraphana">Open in Grafana</button>
+    </div>
+  </li>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { Component, Prop } from 'vue-property-decorator';
+
+import Alert from '@/interfaces/Alert';
+
+@Component
+export default class AlertLine extends Vue {
+  // region properties
+  @Prop(Object) public alert!: Alert;
+  @Prop(String) public topic!: string;
```

Comment:
I would move the props to the `@Component`-Part for consistency with the other components.

Reply:
I disagree, this way to add props to a components is preferrable because it is more readable and offers the possibility annotate props with their exact type.

Example:

```
// props using decorators with exact type annotation
@Prop(Object) public alert!: Alert;
```

```
// props inside component object lose their exact type annotation
@Component ({
    props: {
        alert: Object,
    },
})
```

Reply:
Then this should be changed, right?: https://github.com/openHPI/monitoring-frontend/pull/8/files#diff-b3f8a7afecce4b53147a2f9679f46d7aR17